flow/project.py

```diff
@@ -655,6 +655,10 @@ def __init__(self, name, options="", group_aggregator=None):
         # decorator placement in terms of `@FlowGroupEntry`, `@aggregator`, or
         # `@operation`.
         self.group_aggregator = group_aggregator
+        # We assign operations to a group using a decorator. In order to identify
+        # whether a function is assigned to a group but not as an operation, we need
+        # to store the names of the functions.
+        self.operations = []
```

Comment:
If we're only using this object to check if an operation name exists, this may as well be a set rather than a list.

Reply:
The `operations` variable here is only used for iterating over.
```
for operation in entry.operations:
  if operation not in self._operations:
      # raise error if not found.
```
The `_operation` variable is an instance variable for the `FlowProject` class which is a set.

Reply:
Got it, I didn't check that code carefully enough. This looks fine then.

Reply:
I think you might be saying different things in this comment thread. It sounds like a `set` is actually a viable replacement here, and is already how the `FlowProject` tracks its operations. I carefully reviewed the code and it appears that all the uses of `FlowOperation._flow_groups` and `FlowGroupEntry._operations` can be replaced by `set` instead of `list`. I made this change in ee92b196694fa3befc519ae7f0f7a7a126a10d1e.